src/main/java/be/seeseemelk/mockbukkit/block/state/BlockStateMock.java

```diff
@@ -293,4 +293,17 @@ public static BlockStateMock mockState(@NotNull Block block)
 			return new BlockStateMock(block);
 		}
 	}
+
+	@Override
+	public boolean equals(Object other) {
+		if(other == null) {return false;}
+		if(!(other instanceof BlockState)) {return false;}
+
+		BlockState otherState = (BlockState) other;
+		if(!(otherState.getType() == getType())) {return false;}
```

Comment:
You can simplify this with `if(otherState.getType() != getType())` instead.

Reply:
Oh yeah sorry I just write weird sometimes.